src/main/java/svc/data/citations/datasources/rejis/RejisCitationDataSource.java

```diff
@@ -0,0 +1,211 @@
+package svc.data.citations.datasources.rejis;
+
+import java.net.URI;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.StringJoiner;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import com.google.common.collect.Lists;
+
+import rx.Observable;
+import svc.data.citations.CitationDataSource;
+import svc.data.citations.datasources.rejis.models.RejisCaseList;
+import svc.data.citations.datasources.rejis.models.RejisFullCitation;
+import svc.data.citations.datasources.rejis.models.RejisPartialCitation;
+import svc.data.citations.datasources.rejis.transformers.RejisCitationTransformer;
+import svc.data.citations.filters.CitationFilter;
+import svc.logging.LogSystem;
+import svc.models.Citation;
+
+@Repository
+public class RejisCitationDataSource implements CitationDataSource {
+
+	@Autowired
+	private RejisConfiguration rejisConfiguration;
+
+	@Autowired
+	private RejisCitationTransformer citationTransformer;
+	
+	@Autowired
+	private CitationFilter citationFilter;
+
+	@Autowired
+	private RestTemplate restTemplate;
+	
+	@Autowired
+	private RejisMunicipalityCodesFactory municipalityCodesFactory;
+
+	static private DateTimeFormatter dobFormatter = DateTimeFormatter.ofPattern("uuuu");
+
+	@Override
+	public List<Citation> getByCitationNumberAndDOB(String citationNumber, LocalDate dob) {
+		List<String> municipalityCodes = municipalityCodesFactory.getAllMunicipalityCodes();
+		RejisCaseList rejisCaseList;
+		List<RejisPartialCitation> rejisPartialCitations = new ArrayList<>();
+		
+		int pageNumber = 0;
+		do{
+			pageNumber++;
+			UriComponentsBuilder builder = getCitationNumberBuilder(pageNumber,citationNumber, municipalityCodes);
+			rejisCaseList = performRestTemplateCall(builder.build().encode().toUri());
+			if (rejisCaseList == null){
+				return Lists.newArrayList();
+			}
+			rejisPartialCitations.addAll(rejisCaseList.CaseIndexRows);
+		}while (rejisCaseList.TotalPages > pageNumber);
+		
+		return citationFilter.Filter(getFullCitations(rejisPartialCitations), dob, null);
+	}
+	
+	private UriComponentsBuilder getCitationNumberBuilder(int pageNumber, String citationNumber, List<String> municipalityCodes){
+		UriComponentsBuilder builder =	 UriComponentsBuilder.fromHttpUrl(rejisConfiguration.getRootUrl()+"/ByTicket")
+				.queryParam("CaseOrTicketNum", citationNumber)
```

Comment:
All of REJIS' variable names should be stores somewhere as constant static strings. It can be in this class if this is the only thing that uses them.